src/app/loadout-builder/mod-utils.ts

```diff
@@ -333,3 +333,19 @@ export function getModAssignments(
 
   return mergedResults;
 }
+
+// TODO (ryan) This is a super lazy way of getting unassigned mods but doing it properly
+// is hard. To do it properly we need to check every possible combination and even if its
+// not a valid fit for the mods, we need to calculate how good it is (least number of unassigned?).
+// This will make all slot independant mods unassigned if it doesn't find a valid fit.
+export function getAssignedAndUnassignedMods(
+  items: DimItem[],
+  mods: PluggableInventoryItemDefinition[],
+  defs: D2ManifestDefinitions | undefined,
+  upgradeSpendTier: UpgradeSpendTier
+): [ReturnType<typeof getModAssignments>, PluggableInventoryItemDefinition[]] {
+  const assignedMods = getModAssignments(items, mods, defs, upgradeSpendTier);
+  const flatAssignedMods = Array.from(assignedMods.values()).flat();
+  const unassignedMods = mods.filter((mod) => !flatAssignedMods.includes(mod));
+  return [assignedMods, unassignedMods];
+}
```

Comment:
This is super lazy I know but I feel like it's low value at the moment and this work has taken me quite a while. I will work on this next but as an interim maybe it's best to get the better assignments out if the code looks good.